src/parameters/mod.ts

```diff
@@ -69,34 +93,42 @@ export const ParameterVariable = <P extends ParameterDefinition>(
 ): ParameterVariableType<P> => {
   let param: ParameterVariableType<P> | null = null;
 
-  if (definition.type instanceof Object) {
-    param = ParameterVariable(
+  if (isCustomType(definition.type)) {
+    return ParameterVariable(
       namespace,
       paramName,
       definition.type.definition,
     );
-  } else if (definition.type === SchemaTypes.object) {
-    if ("properties" in definition) {
-      param = CreateTypedObjectParameterVariable(
+  }
+  switch (definition.type) {
+    case SchemaTypes.object:
+      if (isTypedObject(definition)) {
+        param = CreateTypedObjectParameterVariable(
+          namespace,
+          paramName,
+          definition,
+        ) as ParameterVariableType<P>;
+      } else {
+        param = CreateUntypedObjectParameterVariable(namespace, paramName);
+      }
+      break;
+    default:
+      param = CreateSingleParameterVariable(
         namespace,
         paramName,
-        definition,
       ) as ParameterVariableType<P>;
-    } else {
-      param = CreateUntypedObjectParameterVariable(namespace, paramName);
-    }
-  } else {
-    param = CreateSingleParameterVariable(
-      namespace,
-      paramName,
-    ) as ParameterVariableType<P>;
   }
-
+  // TODO: the following type assertion seems to ignore the case where param could be null
```

Comment:
Can we do anything for this todo?

Reply:
Can we chalk it up as something to tackle later and not block this PR anymore? I can't fully remember what the coment is about to be honest

Reply:
Heh, I also didn't follow what the comment was concerned about which is why I was hoping we could resolve before merging 😅 Fixing it by clarifying or removing isn't a PR blocker but would be nice to resolve rather than leaving it in here forever without clarifying

Reply:
I removed it in 30d8fef - the initializing variable at the top of this method initially assigns `null` to it but with the way the logic works out now, it will never remain `null`, so I think it is safe to remove.